pkg/kev/converter/kubernetes/project_service.go

```diff
@@ -552,36 +552,89 @@ func (p *ProjectService) ports() []composego.ServicePortConfig {
 
 // healthcheck returns project service healthcheck probe
 func (p *ProjectService) healthcheck() (*v1.Probe, error) {
+	probeType, err := p.livenessProbeType()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get a valid liveness probe type")
+	}
 
-	if !p.livenessProbeDisabled() {
-		command := p.livenessProbeCommand()
-		timoutSeconds := p.livenessProbeTimeout()
-		periodSeconds := p.livenessProbeInterval()
-		initialDelaySeconds := p.livenessProbeInitialDelay()
-		failureThreshold := p.livenessProbeRetries()
+	var handler v1.Handler
 
-		if len(command) == 0 || len(command[0]) == 0 || timoutSeconds == 0 || periodSeconds == 0 ||
-			initialDelaySeconds == 0 || failureThreshold == 0 {
+	switch probeType {
+	case ProbeTypeNone:
+		return nil, nil
+	case ProbeTypeCommand:
+		handler.Exec = &v1.ExecAction{
+			Command: p.livenessProbeCommand(),
+		}
+
+		if handler.Exec == nil || len(handler.Exec.Command) == 0 || len(handler.Exec.Command[0]) == 0 {
 			log.Error("Health check misconfigured")
 			return nil, errors.New("Health check misconfigured")
 		}
-
-		probe := &v1.Probe{
-			Handler: v1.Handler{
-				Exec: &v1.ExecAction{
-					Command: command,
-				},
-			},
-			TimeoutSeconds:      timoutSeconds,
-			PeriodSeconds:       periodSeconds,
-			InitialDelaySeconds: initialDelaySeconds,
-			FailureThreshold:    failureThreshold,
+	case ProbeTypeHTTP:
+		hprobe, err := p.livenessHTTPProbe()
+		if err != nil {
+			return nil, err
 		}
+		handler.HTTPGet = &hprobe
+	default:
+		return nil, errors.Errorf("unsupported probe type: %s", probeType)
+	}
 
-		return probe, nil
+	timeoutSeconds := p.livenessProbeTimeout()
+	periodSeconds := p.livenessProbeInterval()
+	initialDelaySeconds := p.livenessProbeInitialDelay()
+	failureThreshold := p.livenessProbeRetries()
+
+	if timeoutSeconds == 0 || periodSeconds == 0 || initialDelaySeconds == 0 || failureThreshold == 0 {
+		log.Error("Health check misconfigured")
+		return nil, errors.New("Health check misconfigured")
 	}
 
-	return nil, nil
+	probe := &v1.Probe{
+		Handler:             handler,
+		TimeoutSeconds:      timeoutSeconds,
+		PeriodSeconds:       periodSeconds,
+		InitialDelaySeconds: initialDelaySeconds,
+		FailureThreshold:    failureThreshold,
+	}
+
+	return probe, nil
+}
+
+func (p *ProjectService) livenessProbeType() (ProbeType, error) {
+	t, ok := p.Labels[config.LabelWorkloadLivenessProbeType]
+	if !ok {
+		return ProbeTypeNone, errors.New("probe type not provided")
+	}
+
+	pt, ok := ProbeTypeFromString(t)
+	if !ok {
+		return ProbeTypeNone, errors.Wrapf(ErrUnsupportedProbeType, "type: %s", t)
```

Comment:
So the problem with the approach of defaulting to something is that we'll end up with something like a misspelled value causing the probe to be disabled so in my opinion we should make it explicit. Returning None was essentially to avoid having something like an Invalid type in the enum because then we'd have to validate whether it's an error AND whether it's an invalid probe type which I thought wasn't really great since by convention you shouldn't use the value if the error is not nil (has error)

Reply:
It's OK to return default `none` probe here. My comment above was around an error being returned... As the (default) value of `ProbeTypeNone` is valid I don't think returning an error here makes much sense hence the original suggestions that it'd be better to warn instead. This warning message could contain information stating that the supplied type was invalid and that default `none` type will be used instead.